htdocs/discover_tab.js

```diff
@@ -0,0 +1,58 @@
+var discover = function () {
+
+    /* "Model-Model-View-Controller"
+     .. or something ..
+     there are two models: the local state and the global rcs+github
+     the local model is private data of this class
+    */
+
+    // local model (all caches of stuff stored in RCS/github)
+    var notebook_info_ = {}, // all notebooks we are aware of
+        num_stars_ = {}, // number of stars for all known notebooks
+        fork_count_ = {};
+
+    function clean_r(obj) { 
+        delete obj.r_attributes; 
+        delete obj.r_type; 
+        return obj; 
+    };
+
+    function load_everything(){
+        var allnotebooks = rcloud.config.get_all_notebook_info();
+
+        var promises = [allnotebooks.then(function(notebooks) {
+            // populate notebook infos
+            notebook_info_ = clean_r(notebooks.infos);
+        }),
+        allnotebooks.then(function(notebooks) {
+            // populate star counts
+            num_stars_ = clean_r(notebooks.stars);
+        }),
+        allnotebooks.then(function(notebooks) {
+            // populate fork counts
+            fork_count_ = clean_r(notebooks.forks);
```

Comment:
That's fine - we need to populate RCS, since this value normally comes from GitHub. You should plan for all the values coming up null initially, and you can populate some values by forking the notebooks of other people (causing an update).

We'll write a notebook that iterates through all current notebooks and populates RCS - but this is intense and will run against GitHub rate limits, so we have to be careful how we do it.